x-pack/plugins/cases/server/connectors/cases/cases_connector.ts

```diff
@@ -125,7 +126,13 @@ export class CasesConnector extends SubActionConnector<
       groupedAlertsWithCaseId
     );
 
-    await this.attachAlertsToCases(casesClient, groupedAlertsWithCases, params);
+    const groupedAlertsWithClosedCasesHandled = await this.handleClosedCases(
+      params,
+      casesClient,
+      groupedAlertsWithCases
+    );
+
+    await this.attachAlertsToCases(casesClient, groupedAlertsWithClosedCasesHandled, params);
```

Comment:
At which point do you check if the cases already have 1K alerts attached to them?

Reply:
good question, should there also be tests for alerts limit error?

Reply:
I plan to handle the 1K limit on another PR. The following PR will be about error handling and retries.